include/index_config.h

```diff
@@ -177,19 +190,20 @@ class IndexConfigBuilder
         if (_data_type == "" || _data_type.empty())
             throw ANNException("Error: data_type can not be empty", -1);
 
-        if (_dynamic_index && _index_write_params != nullptr)
+        if (_dynamic_index && _num_frozen_pts == 0)
         {
-            if (_search_threads == 0)
-                throw ANNException("Error: please pass search_threads for building dynamic index.", -1);
+            _num_frozen_pts = 1;
+        }
 
-            if (_initial_search_list_size == 0)
+        if (_dynamic_index && _index_search_params != nullptr)
+        {
+            if (_index_search_params->initial_search_list_size == 0)
```

Comment:
Shouldn't this check be present for both _dynamic and _static indices?

Reply:
I see, i will add that.